src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWrite.cs

```diff
@@ -20,43 +20,37 @@ internal enum ColorUInt64 : ulong { Red, Blue, Green };
 
     public partial class TestsWrite
     {
-
         [EventData]
         private struct PartB_UserInfo
         {
             public string UserName { get; set; }
         }
 
-        /// <summary>
-        /// Tests the EventSource.Write[T] method (can only use the self-describing mechanism).
-        /// Tests the EventListener code path
-        /// </summary>
-        [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/21564", TargetFrameworkMonikers.NetFramework)]
-        public void Test_Write_T_EventListener()
+        public static TheoryData<Listener> GetListeners()
         {
-            using (var listener = new EventListenerListener())
+            TheoryData<Listener> data = new TheoryData<Listener>();
+            if (PlatformDetection.IsNetCore && PlatformDetection.IsNotAndroid && PlatformDetection.IsNotBrowser && PlatformDetection.IsNotMonoRuntime)
```

Comment:
I guess EventPipe tests would work on Mono runtime as well, excluded just because it was not tested or is there any know issues or limitations?

Reply:
I excluded it because it appeared `DiagnosticClient(int pid)` wasn't working to self-attach. Does that sound right or maybe I misinterpreted something?